### PR TITLE
ChainIDE + Polkaholic.io WASM Contract Explorer Integration - Milestone 1 delivery

### PR DESCRIPTION
**Pull request by sourabhniyogi**

### CODE REVIEW

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+
+### ChainIDE-Polkaholic Verification Build Process: from Milestone 1 to Milestone 2
+
+ChainIDE uses a containerized build process, where tiny variations in configuration may result in different codehashes.  This is acceptable for the above security model where ChainIDE is currently assumed to be a trusted IDE.   However, to standardize WASM Contract compilation such that multiple parties (in particular Colorful Notion and ChainIDE as well as others) can use identical build processes to derive the exact same code hash, in July 2023 Parity released the following  "cargo contract build --verifiable" functionality:
+**[https://github.com/paritytech/cargo-contract/releases/tag/v4.0.0-alpha](https://github.com/paritytech/cargo-contract/releases/tag/v4.0.0-alpha)**
+The official version is expected to be launched by the end of the year.  Both teams researched the usability of this (ChainIDE to incorporate into its build process, Polkaholic to replicate and arrive).   See issue [here](TODO).
+
```
Comment (sourabhniyogi) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:66]: @wufengtao1 Add issue here.

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+Both teams will complete Milestone 1 in September and are developing plans to propose Milestone 2 in November, after using October as a feedback collection period.  
+
+Here is the detailed timeline:
+* September 2023 - ChainIDE is awaiting QA from Colorful Notion while Astar is currently undergoing acceptance testing.  Colorful Notion is completing this document and finalizing functionality for a developer preview in October 2023.
+
+* *October 2023* - ChainIDE + Colorful Notion  request feedback from curators, all inkubator WASM Contract Developers and experienced WASM Contract Developers to:
+
+	1. Try ChainIDE at [https://staging-9589904a8a.chainide.com/s/dashboard/projects](https://staging-9589904a8a.chainide.com/s/dashboard/projects) using Shibuya and the SBY faucet.   We recommend using PSP22, PSP34 or the UniswapV2 contract templates, and then using your own WASM Contract.  
+
+	2. For bug reports and feature requests, submit them [here](TODO) or join our open group here [ChainIDE+Colorful Notion+Astar]()
```
Comment (sourabhniyogi) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:186]: @wufengtao1 Add links here

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+## 0c. Testing and Testing Guide
+
+The ChainIDE-Polkaholic unit tests are comprehensively demonstrated in the following video link:
+
+[Comprehensive ChainIDE Walkthrough, including Polkaholic.io integration (Video Link)](https://www.dropbox.com/scl/fi/tf0jgxsmss5b52uk0nbw7/Astar-ChainIDE-M2-New.mp4?rlkey=ts3tpvzvxpbfg7wrl7f4va1yl&dl=0)
+
+Any WASM Contract developer can preview:
+
+[ChainIDE WASM Contract Development Preview - September 2023](https://staging-9589904a8a.chainide.com/s/dashboard/projects) (Note: to be finalized in October!)
```
Comment (sourabhniyogi) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:126]: @wufengtao1 Adjust links as appropriate

**Comment by HCastano** [comment]
Hey, we haven't had time to review this yet, but we'll try to soon 🙇

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+### Polkaholic.io API for WASM Contracts
+
+The [Polkaholic.io API](https://docs.polkaholic.io/#introduction) has been extended to include the following WASM Contract APIs at the same REST API Endpoint of [https://api.polkaholic.io](https://api.polkaholic.io) to enable anyone to access data about a network and contract address with simple `{network}` and `{contractAddress}` parameters.
+
+|Function|REST API Format|Example|
+|---|---|--|
+|Retrieve indexed code hashes by code hash | https://api.polkaholic.io/wasmcode/`{network}` | https://api.polkaholic.io/wasmcode/shiden |
+|Retrieve contract addresses | https://api.polkaholic.io/wasmcontracts/`{network}` | https://api.polkaholic.io/wasmcontracts/shibuya |
+|  Retrieve contract calls of a contract address |  https://api.polkaholic.io/wasmcontract/`{network}`/`{contractAddress}`  | https://api.polkaholic.io/wasmcontract/shiden/XRcGZzdH841dHySiM62BfJDbQfXXYex3U7LYooRJt8EgJeX |
```
Comment (DoubleOTheven) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:81]: This example URL is throwing a 404 error

```Cannot GET /wasmcontract/shiden/XRcGZzdH841dHySiM62BfJDbQfXXYex3U7LYooRJt8EgJeX```

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+ChainIDE integrated both of the above dashboards with IFRAMEs and used the CSS customization capabilities to match its Dark Mode.
+
+It is expected that additional dashboards can be done for other purposes (e.g. a line chart showing # of calls, contracts verified, etc. by day/month/quarter, by network etc.) for other contexts.  The same Superset can be used to expose these in Polkaholic.io, as well support many other blockchain analytics problems.    
+
+## 0c. Testing and Testing Guide
+
+The ChainIDE-Polkaholic unit tests are comprehensively demonstrated in the following video link:
+
+[Comprehensive ChainIDE Walkthrough, including Polkaholic.io integration (Video Link)](https://www.dropbox.com/scl/fi/tf0jgxsmss5b52uk0nbw7/Astar-ChainIDE-M2-New.mp4?rlkey=ts3tpvzvxpbfg7wrl7f4va1yl&dl=0)
```
Comment (DoubleOTheven) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:122]: Super cool 😎

**Review by DoubleOTheven — APPROVED** [review]
This looks very nice. Thanks for the easy-to-follow updates, links, and videos

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 4 above)
Comment (HCastano) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:81]: Ya this is still an issue for me

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+*  WASM Contract UI from Polkaholic.io
+
+WASM Contract Developer feedback (new features) will be requested in October 2023 from Milestone 1 Deliverables and used to draft a Milestone 2 follow-on proposal.
+| Number | Deliverable | Link | Notes |
+| --- | --- | --- | --- |
+| 0a. | License | [Polkaholic GNU GPL v3](https://github.com/colorfulnotion/polkaholic/blob/main/LICENSE) |   |
+| 0b. | Polkaholic Documentation | See 0b. Polkaholic Documentation below  | See Polkaholic screenshots and URLs |
+| 0c. | Testing and Testing Guide | See 0c. below | See ChainIDE video walkthrough |
+| 1. | Repository | https://github.com/colorfulnotion/polkaholic  |  |
+| 2. | Repository | https://staging-9589904a8a.chainide.com/s/dashboard/projects | ChainIDE is only semi-open source. This is [the open-source plugin template for ChainIDE](https://github.com/WhiteMatrixTech/chainide-plugin-doc). If you need other open-source plugins, please let us know. |
```
Comment (HCastano) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:32]: Can you link to some of the other ink! relevant components of ChainIDE? It's unclear to me what's open-source and what's not

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 1 above)
Comment (HCastano) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:66]: Looks like this PR has been merged

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 5 above)
Comment (HCastano) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:122]: Ya!

One piece of advice here. I'd split the video out into a series of shorter (1-3min) videos to make it easier for people to find what they need quickly.

I'd also like to see this on YouTube or something more accessible

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+Both teams will complete Milestone 1 in September and are developing plans to propose Milestone 2 in November, after using October as a feedback collection period.  
+
+Here is the detailed timeline:
+* September 2023 - ChainIDE is awaiting QA from Colorful Notion while Astar is currently undergoing acceptance testing.  Colorful Notion is completing this document and finalizing functionality for a developer preview in October 2023.
+
+* *October 2023* - ChainIDE + Colorful Notion  request feedback from curators, all inkubator WASM Contract Developers and experienced WASM Contract Developers to:
+
+	1. Try ChainIDE at [https://staging-9589904a8a.chainide.com/s/dashboard/projects](https://staging-9589904a8a.chainide.com/s/dashboard/projects) using Shibuya and the SBY faucet.   We recommend using PSP22, PSP34 or the UniswapV2 contract templates, and then using your own WASM Contract.  
+
+	2. For bug reports and feature requests, submit them [here](TODO) or join our open group here [ChainIDE+Colorful Notion+Astar]()
```
Comment (HCastano) [inline deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:186]: There's a TODO linked here

**Review by HCastano — APPROVED** [review]
Great submission!

Thanks for laying out all the features and providing links and videos, super helpful during my review!

What are the blockers before this can be put into production? I'd love to see developers start to leverage the IDE as well as all the APIs behind it

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 4 above)
Comment (sourabhniyogi) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md]: Apologies, this is fixed -- url is actually 
 
 https://api.polkaholic.io/wasmcontract/shibuya/XRcGZzdH841dHySiM62BfJDbQfXXYex3U7LYooRJt8EgJeX

**Comment by TtomaS7** [comment]
Hey team!

Here's the form for payment details: https://forms.gle/BUCKfygkiEvpdYna6.
Once you've filled it out, please let me know and be sure to specify what it pertains to; you can include the link to this PR delivery! Further details can be provided under the section: "Please describe which Milestone you have completed and attach the link to the Milestone delivery."

Should you have any questions, don't hesitate to ask. Thank you so much!

Important Note: PLEASE VERIFY YOUR ADDRESS, WHERE YOU WILL RECEIVE THE FUNDS, BEFORE FILLING OUT THE FORM!
For guidance on how to set up identity, refer to [this guide](https://support.polkadot.network/support/solutions/articles/65000181981-how-to-set-and-clear-an-identity).

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 6 above)
Comment (wufengtao1) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:32]: Hi, I come from ChainIDE, and as of now, the only open-source component is the open-source plugin template, allowing users to develop plugins on ChainIDE. However, {the other modules](https://chainide.gitbook.io/chainide-english-1/chainide-modules), including the backend, are not open-source. If you need other parts to be open-source as well, I believe we can discuss the possibility further.

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 5 above)
Comment (wufengtao1) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:122]: This is a good suggestion, and I believe I can have it completed by next week.

**Comment by wufengtao1** [comment]
> Hey team!
> 
> Here's the form for payment details: https://forms.gle/BUCKfygkiEvpdYna6. Once you've filled it out, please let me know and be sure to specify what it pertains to; you can include the link to this PR delivery! Further details can be provided under the section: "Please describe which Milestone you have completed and attach the link to the Milestone delivery."
> 
> Should you have any questions, don't hesitate to ask. Thank you so much!
> 
> Important Note: PLEASE VERIFY YOUR ADDRESS, WHERE YOU WILL RECEIVE THE FUNDS, BEFORE FILLING OUT THE FORM! For guidance on how to set up identity, refer to [this guide](https://support.polkadot.network/support/solutions/articles/65000181981-how-to-set-and-clear-an-identity).

Hi, Tomo. I come from ChainIDE. According to the contract, the grant amount is allocated as follows: 120,000 USD Total: 60,000 USD (Colorful Notion), 60,000 USD (ChainIDE). Does this mean that both Colorful and ChainIDE need to fill out this form?

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 7 above)
Comment (sourabhniyogi) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md]: We have adjusted it to point to the inkubator matrix room.  If you have a better idea, pls advise.

**Comment by sourabhniyogi** [comment]
> Hi, Tomo. I come from ChainIDE. According to the contract, the grant amount is allocated as follows: 120,000 USD Total: 60,000 USD (Colorful Notion), 60,000 USD (ChainIDE). Does this mean that both Colorful and ChainIDE need to fill out this form?

@wufengtao1 We did our part [for Colorful Notion], you can do your part [for ChainIDE] -- thanks!

**Comment by wufengtao1** [comment]
@TtomaS7 @sourabhniyogi  ChainIDE has submitted the form for payment details, and the email is  wufengtao2@gmail.com :)

**Comment by sourabhniyogi** [comment]
> Hey team!
> 
> Here's the form for payment details: https://forms.gle/BUCKfygkiEvpdYna6. Once you've filled it out, please let me know and be sure to specify what it pertains to; you can include the link to this PR delivery! Further details can be provided under the section: "Please describe which Milestone you have completed and attach the link to the Milestone delivery."
> 
> Should you have any questions, don't hesitate to ask. Thank you so much!
> 
> Important Note: PLEASE VERIFY YOUR ADDRESS, WHERE YOU WILL RECEIVE THE FUNDS, BEFORE FILLING OUT THE FORM! For guidance on how to set up identity, refer to [this guide](https://support.polkadot.network/support/solutions/articles/65000181981-how-to-set-and-clear-an-identity).

As per your request, we resubmitted with just 1 address to receive payment for both teams.  Thank you!

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 5 above)
Comment (wufengtao1) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:122]: > Ya!
> 
> One piece of advice here. I'd split the video out into a series of shorter (1-3min) videos to make it easier for people to find what they need quickly.
> 
> I'd also like to see this on YouTube or something more accessible


By the way, do you think using the original video directly is sufficient, or do we need to optimize the original video to meet the specifications for the promotional video?

**Comment by TtomaS7** [comment]
You can claim your rewards in 7 days!!
https://polkadot.subsquare.io/treasury/child-bounties/19_462

**Comment by sourabhniyogi** [comment]
> Great submission!
> 
> Thanks for laying out all the features and providing links and videos, super helpful during my review!
> 
> What are the blockers before this can be put into production? I'd love to see developers start to leverage the IDE as well as all the APIs behind it

@HCastano  Can you suggest a non-Astar production path that we can explore with you by the end of the year, which we can address in Milestone 2.  Ready to have a "next steps" discussion.

We need to get coordinated on the verification process, PolkaVM, Rococo => ____, and the basic "Who are the top 10 WASM Contract devs who build what that users actually use" question?

File: deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md (hunk 5 above)
Comment (wufengtao1) [reply deliveries/chainide-polkaholic-integrated-wasm-contract-explorer-integration-milestone_1.md:122]: > Ya!
> 
> One piece of advice here. I'd split the video out into a series of shorter (1-3min) videos to make it easier for people to find what they need quickly.
> 
> I'd also like to see this on YouTube or something more accessible

https://www.youtube.com/watch?v=q-Bf5RHSt4s

**Comment by HCastano** [comment]
@sourabhniyogi Hey sorry, just saw your last message. Is this something that you still need help with?

**Comment by sourabhniyogi** [comment]
I believe we should chart a path for ink!ubator for Coreplay and get coordinated on that.

**Pull request merged**